### PR TITLE
test: Tracked accuracy of categorization scheme through fixtures

### PR DESCRIPTION
**Pull request by francoisWeber**
Added a new test in order to have human-readable accuracy tracking of the categorization scheme.
The PR contains:
- update of `package.json` and adding of `Makefile` : added a few commands to fetch/decrypt/decrypt/clean the fixtures needed for the test
- added `allowed_wrong_categories.json` that stores the allowed "mistakes" between alike categories (example : "snacks, meal at work" could reasonably be "restaurant, bar")
- added a test `services-fixtures.spec.js` that fetches the last categorization model from Cozy, available banking fixtures and applies the categorization scheme to every operations. It also measures the per-fixture accuracy and the global accuracy of the categorization scheme

### CODE REVIEW

File: src/ducks/categorization/services-fixtures.spec.js [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+      `${bank}-clean-transactions.bi.json`
+    )
+    let transactions
+    try {
+      transactions = require(expectedPath)['io.cozy.bank.operations']
+    } catch (error) {
+      transactions = undefined
+    }
+    // const transactions = require(expectedPath)["io.cozy.bank.operations"];
+    // if fixture exists : continue
```
Comment (ptbrowne) [inline src/ducks/categorization/services-fixtures.spec.js]: Forgotten comments ?

Comment (francoisWeber) [reply src/ducks/categorization/services-fixtures.spec.js]: ups, my mistake

**Comment by francoisWeber** [comment]
Edit: Removed the Promise system because the testing order appears to be proper

**Comment by francoisWeber** [comment]
EDIT 25/02:
I added a way of outputting the results in CSV + txt in a folder given in the file `src/ducks/categorization/local-config.json`. This was done in order to easily share these metrics with  Banks' product owner via a dedicated shared folder of our infrastructure.

**Comment by francoisWeber** [comment]
Resolved conflicts in `packages.json`

File: package.json [hunk 2]
```diff
@@ -165,11 +170,13 @@
     "cozy-pouch-link": "5.7.6",
     "cozy-stack-client": "5.6.1",
     "cozy-ui": "19.3.0",
+    "csv-writer": "^1.2.0",
```
Comment (kosssi) [inline package.json]: can you fix version? (remove ^ and execute `yarn`)

File: package.json [hunk 3]
```diff
@@ -137,6 +141,7 @@
     "stylus": "0.54.5",
     "stylus-loader": "3.0.2",
     "svg-sprite-loader": "3.9.2",
+    "tar": "4.4.8",
```
Comment (kosssi) [inline package.json]: Tu n'as pas besoin de cette dépendance non plus

File: package.json [hunk 4]
```diff
@@ -165,11 +170,13 @@
     "cozy-pouch-link": "5.7.6",
     "cozy-stack-client": "5.6.1",
     "cozy-ui": "19.3.0",
+    "csv-writer": "^1.2.0",
     "d3": "5.9.1",
     "date-fns": "1.30.1",
     "detect-node": "2.0.4",
     "expose-loader": "0.7.5",
     "fastclick": "1.0.6",
+    "fs": "^0.0.1-security",
```
Comment (kosssi) [inline package.json]: tu n'as pas besoin non plus de celle ci ?

File: src/ducks/categorization/.gitignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+fixtures/
+bank_classifier_nb_and_voc.json
+__snapshots__/
+local-config.json
```
Comment (kosssi) [inline src/ducks/categorization/.gitignore]: il manque un saut de ligne ;)

File: package.json (hunk 2 above)
Comment (francoisWeber) [reply package.json]: Si, j'en ai besoin dans `service-fixture.spec.js` ligne 63

File: package.json (hunk 3 above)
Comment (francoisWeber) [reply package.json]: Ça effectivement c'est un artefact

File: package.json (hunk 4 above)
Comment (francoisWeber) [reply package.json]: Si, j'en ai besoin dans `service-fixture.spec.js` ligne 476

File: src/ducks/categorization/.gitignore (hunk 5 above)
Comment (francoisWeber) [reply src/ducks/categorization/.gitignore]: J'ignorais qu'il en fallait un dans les `.gitignore`; j'ajoute ça !

File: src/ducks/categorization/allowed_wrong_categories.json [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+  "400200": [
+    "400200"
+  ],
+  "400800": [
+    "400800"
+  ],
+  "0": [
+    "0"
+  ]
+}
```
Comment (kosssi) [inline src/ducks/categorization/allowed_wrong_categories.json]: il manque un saut de ligne

File: package.json [hunk 7]
```diff
@@ -56,6 +56,10 @@
     "styleguide": "styleguidist server",
     "styleguide:build": "styleguidist build",
     "styleguide:deploy": "git-directory-deploy --directory ./docs/build/styleguide --branch master --repo docs",
+    "encrypt-banking-tests": "cd src/ducks/categorization/; make encrypted.tar.gz.gpg",
+    "clean-banking-tests": "cd src/ducks/categorization/; make clean",
+    "decrypt-banking-tests": "cd src/ducks/categorization; gpg -d encrypted.tar.gz.gpg | tar xz",
+    "download-banking-tests": "cd src/ducks/categorization; wget https://github.com/cozy/cozy-libs/raw/master/packages/cozy-doctypes/src/banking/encrypted.tar.gz.gpg -O encrypted.tar.gz.gpg ; wget https://bank-autocategorization.cozycloud.cc/bank_classifier_nb_and_voc.json -O bank_classifier_nb_and_voc.json"
```
Comment (kosssi) [inline package.json]: il manque un `,` à la fin de la ligne

File: package.json (hunk 2 above)
Comment (francoisWeber) [reply package.json]: Je n'avais pas vu ton commentaire pour `devDependencies, je change

File: package.json (hunk 4 above)
Comment (francoisWeber) [reply package.json]: `fs` supprimé des dépendences

File: package.json (hunk 2 above)
Comment (francoisWeber) [reply package.json]: Suppression de `^`

**Pull request merged**